storage-resize-images/functions/src/resize-image.ts

```diff
@@ -0,0 +1,181 @@
+  fileDir: string;
+  fileNameWithoutExtension: string;
+  fileExtension: string;
+  contentType: string;
+  size: string;
+  objectMetadata: ObjectMetadata;
+  remoteFile: File;
+  filePath: string;
+}): Promise<ResizedImageResult> => {
+  const { imageType } = config;
+  const hasImageTypeConfigSet = imageType !== "false";
+  const imageContentType = hasImageTypeConfigSet
+    ? supportedImageContentTypeMap[imageType]
+    : contentType;
+  const modifiedExtensionName =
+    fileExtension && hasImageTypeConfigSet ? `.${imageType}` : fileExtension;
+  const modifiedFileName = `${fileNameWithoutExtension}_${size}${modifiedExtensionName}`;
+  // Path where modified image will be uploaded to in Storage.
+  const modifiedFilePath = path.normalize(
+    config.resizedImagesPath
+      ? path.join(fileDir, config.resizedImagesPath, modifiedFileName)
+      : path.join(fileDir, modifiedFileName)
+  );
+  let modifiedFile: string;
+
+  try {
+    modifiedFile = path.join(os.tmpdir(), modifiedFileName);
+
+    // Cloud Storage files.
+    const metadata: { [key: string]: any } = {
+      contentDisposition: objectMetadata.contentDisposition,
+      contentEncoding: objectMetadata.contentEncoding,
+      contentLanguage: objectMetadata.contentLanguage,
+      contentType: imageContentType,
+      metadata: objectMetadata.metadata || {},
+    };
+    metadata.metadata.resizedImage = true;
+    if (config.cacheControlHeader) {
+      metadata.cacheControl = config.cacheControlHeader;
+    } else {
+      metadata.cacheControl = objectMetadata.cacheControl;
+    }
+
+    // If the original image has a download token, add a
+    // new token to the image being resized #323
+    if (metadata.metadata.firebaseStorageDownloadTokens) {
+      metadata.metadata.firebaseStorageDownloadTokens = uuid();
+    }
+
+    // Generate a resized image buffer using Sharp.
+    logs.imageResizing(modifiedFile, size);
+    let modifiedImageBuffer = await resize(originalFile, size);
+    logs.imageResized(modifiedFile);
+
+    // Generate a converted image type buffer using Sharp.
+    if (hasImageTypeConfigSet) {
+      logs.imageConverting(fileExtension, config.imageType);
+      modifiedImageBuffer = await convertType(modifiedImageBuffer);
+      logs.imageConverted(config.imageType);
+    }
+
+    // Generate a image file using Sharp.
+    await sharp(modifiedImageBuffer).toFile(modifiedFile);
+
+    // Uploading the modified image.
+    logs.imageUploading(modifiedFilePath);
+    await bucket.upload(modifiedFile, {
+      destination: modifiedFilePath,
+      metadata,
+    });
+    logs.imageUploaded(modifiedFile);
+
+    if (config.deleteOriginalFile === deleteImage.onSuccess) {
+      if (remoteFile) {
+        try {
+          logs.remoteFileDeleting(filePath);
+          await remoteFile.delete();
+          logs.remoteFileDeleted(filePath);
+        } catch (err) {
+          logs.errorDeleting(err);
+        }
+      }
+    }
+
+    return { size, success: true };
```

Comment:
since we pass the status back anyway, what do you think about doing the delete back in index.ts, just so the two deletes (`always` and `onSuccess`) are closer to each other in the code?